src/controller/python/OpCredsBinding.cpp

```diff
@@ -700,4 +710,17 @@ PyChipError pychip_GetCompletionError()
     return ToPyChipError(sTestCommissioner.GetCompletionError());
 }
 
+PyChipError pychip_DeviceController_SetDACRevocationSetPath(const char * dacRevocationSetPath)
+{
+    chip::Credentials::DeviceAttestationRevocationDelegate * dacRevocationDelegate =
+        GetTestAttestationRevocationDelegate(dacRevocationSetPath);
+    VerifyOrReturnError(dacRevocationDelegate != nullptr, ToPyChipError(CHIP_ERROR_INVALID_ARGUMENT));
+
+    chip::Credentials::DefaultDACVerifier * dacVerifier = chip::Credentials::GetDefaultDACVerifierInstance();
```

Comment:
I am not following why you cannot get `DefaultDACVerifier *` from `GetDefaultDACVerifier()`. Something seems off to me here. I know you mentioned that `dynamic_cast` didn't work, but my follow up question is why didn't it work. It really seems like we are putting up a bandaid to fix something that we should understand why the correct approach is not working. And if there is a legitimate reason why that isn't possible we need to communicate that in comment so the next person reviewing this section of code understand why this approach was taken

Reply:
Looking at this a little more, would it make sense for `DeviceAttestationVerifier` to have a virtual function `CHIP_ERROR TrySetRevocationDelegate` that by default returns `CHIP_ERROR_NOT_IMPLEMENTED`, and in `DefaultDACVerifier` you override it to do what you want?

Reply:
`dynamic_cast` is not possible because because of `-fno-rtti` flag is set when building the python controller. ( I see few failures in CI)
On adding that flag, we don't get the run time type information which stops us from performing the `dynamic_cast`. But, we get some space savings.

Reply:
Sorry, that I did not mention the reason why `dynamic_cast` was failing, I should have.

Reply:
I did gave a though about adding `SetRevocationDelegate()` into `DeviceAttestationVerifier`, but after looking into current members of [DeviceAttestationVerifier](https://github.com/project-chip/connectedhomeip/blob/master/src/credentials/attestation_verifier/DeviceAttestationVerifier.h#L263), I don't think its a right place to have add that API.

Reply:
Why doesn't is seem like an okay place for it? Doesn't `DeviceAttestationVerifier` already have a pure virtual method for `CheckForRevokedDACChain`?

Reply:
I think not adding a `SetRevocationDelegate()` there keeps that interface clean and `CheckForRevokedDACChain()` can be implemented without need of revocation delegate.

If we add `SetRevocationDelegate()` there, then we would be forcing the derived implementations to have that method and implement it somehow, even if they don't want to use that interface.

Adding a `Set_` method which does nothing may not be a good thing, IMO.

How I am seeing the `DefaultDACVerifier` is it implements methods in `DeviceAttestationVerifier` + introduces the test implementations of paa-trust-store and dac-revocation-set.

Reply:
Sorry I am not following how something like the following is "forcing derived implementations to have a method", you are allowing implementations to optionally provide that ability:

```
virtual CHIP_ERROR TrySettingRevocationDelegate(DeviceAttestationRevocationDelegate * revocationDelegate)
{
    return CHIP_ERROR_NOT_IMPLEMENTED
}
```

Reply:
Thanks @tehampson for clarifying that adding `TrySettingRevocationDelegate()` is better than the approach that I used earlier.

I have force pushed the branch, and re-wrote the history in order to make it cleaner.

Please re-review the changes.